homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -764,11 +792,20 @@ def __init__(self, name, device, entry, unique_id, coordinator):
         else:
             self._percentage = self.coordinator.data.direct_speed
 
+    @property
+    def operation_mode_class(self):
+        """Hold operation mode class."""
+
     @property
     def preset_mode(self):
         """Get the active preset mode."""
         return ATTR_MODE_NATURE if self._nature_mode else ATTR_MODE_NORMAL
 
+    @property
+    def preset_modes(self) -> list:
+        """Get the list of available preset modes."""
+        return [ATTR_MODE_NATURE, ATTR_MODE_NORMAL]
```

Comment:
Perhaps you could use self._attr_preset_modes in __init__

Reply:
I think fan platform doesn't support attribute shorthand for property implementation.

Reply:
Ok